### PR TITLE
Fix multiple with port calls

### PR DESCRIPTION
**Pull request by elivlo**
Hi,

I modified the function "WithPorts" for type Scanner. This will get around the error:
`[Only 1 -p option allowed, separate multiple ranges with commas. QUITTING! EOF]`

Also modified _nmap_test.go_ to check if the new function works.

best regards

### CODE REVIEW

File: nmap.go [hunk 1]
```diff
@@ -670,8 +670,23 @@ func WithPorts(ports ...string) func(*Scanner) {
 	portList := strings.Join(ports, ",")
 
 	return func(s *Scanner) {
-		s.args = append(s.args, "-p")
-		s.args = append(s.args, portList)
+		//find if any port set
```
Comment (Ullaakut) [inline nmap.go:673]:
```suggestion
		// Find if any port is set.
```

File: nmap.go [hunk 2]
```diff
@@ -670,8 +670,23 @@ func WithPorts(ports ...string) func(*Scanner) {
...
+		var place int = -1
+		for p, value := range s.args {
+			if value == "-p" {
+				place = p
+				break
+			}
+		}
+
+		//add ports
```
Comment (Ullaakut) [inline nmap.go:682]:
```suggestion
		// Add ports.
```

**Review by Ullaakut — APPROVED** [review]

**Comment by Ullaakut** [comment]
Thank you @elivlo :)))

**Comment by elivlo** [comment]
My pleasure! :-)

**Review by elivlo — COMMENTED** [review]
Good advice!

**Pull request merged**